editor/src/components/canvas/controls/canvas-offset-wrapper.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { CanvasVector } from '../../../core/shared/math-utils'
+import { useRefEditorState, useSelectorWithCallback } from '../../editor/store/store-hook'
+
+export const CanvasOffsetWrapper = React.memo((props) => {
+  const elementRef = useApplyCanvasOffset(null)
+
+  return (
+    <div ref={elementRef} style={{ position: 'absolute' }}>
+      {props.children}
+    </div>
+  )
+})
+
+export function useApplyCanvasOffset(scale: number | null): React.RefObject<HTMLDivElement> {
```

Comment:
I think for future debugging, the name of this hook should tell us that it overrides the style.transform prop of the ref

Reply:
fixed